ansible/library/conn_graph_facts.py

```diff
@@ -231,28 +228,63 @@ def get_host_connections(self, hostname):
         else:
             return self.links
 
+    def contains_hosts(self, hostnames):
+        return set(hostnames) <= set(self.devices)
+
+
+LAB_CONNECTION_GRAPH_FILE = 'graph_files.yml'
+EMPTY_GRAPH_FILE = 'empty_graph.xml'
+LAB_GRAPHFILE_PATH = 'files/'
+
+"""
+    Find a graph file contains all devices in testbed.
+"""
+def find_graph(hostnames, anchor):
+    filename = LAB_GRAPHFILE_PATH + LAB_CONNECTION_GRAPH_FILE
+    with open(filename) as fd:
+        file_list = yaml.load(fd)
```

Comment:
I think pyyaml will complain about this when we flip to python3, did you try using `safe_load` instead?

Reply:
I didn't try. I prefer to get this merged as-is since it is tested. Then improve.